### PR TITLE
refactor(misconf): Deprecate `EXCEPTIONS` for misconfiguration scanning

### PR DESCRIPTION
**Pull request by simar7**
## Description

Deprecate `EXCEPTIONS` for misconfiguration scanning

### Before
```shell

main.tf (terraform)

Tests: 8 (SUCCESSES: 0, FAILURES: 7, EXCEPTIONS: 1)
Failures: 7 (UNKNOWN: 0, LOW: 2, MEDIUM: 1, HIGH: 4, CRITICAL: 0)

HIGH: No public access block so not blocking public acls
```

### After
```shell
main.tf (terraform)

Tests: 7 (SUCCESSES: 0, FAILURES: 7)
Failures: 7 (UNKNOWN: 0, LOW: 2, MEDIUM: 1, HIGH: 4, CRITICAL: 0)

HIGH: No public access block so not blocking public acls
```

NB Ignores are still listed as part of the regular log output
```
2024-10-23T13:46:30-06:00       INFO    [terraform executor] Ignore finding     rule="aws-s3-encryption-customer-key" range="main.tf:13-21"
```

## Related stuff 
- https://github.com/aquasecurity/trivy/discussions/7474
- Close https://github.com/aquasecurity/trivy/issues/7314

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [x] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/flag/misconf_flags.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ var (
 	IncludeNonFailuresFlag = Flag[bool]{
 		Name:       "include-non-failures",
 		ConfigName: "misconfiguration.include-non-failures",
-		Usage:      "include successes and exceptions, available with '--scanners misconfig'",
+		Usage:      "include successes and failures, available with '--scanners misconfig'",
```
Comment (nikpivkin) [inline pkg/flag/misconf_flags.go:37]: Failures are enabled by default, this flag additionally enables successes

File: pkg/rpc/convert.go
Comment (DmitriyLewen) [inline pkg/rpc/convert.go:1]: IIUC we can mark `exceptions` as `reserved`:
https://github.com/aquasecurity/trivy/blob/4926da79de901fba73819d71845ec0355b68ae0f/rpc/common/service.proto#L76

File: contrib/junit.tpl [hunk 2]
```diff
@@ -16,7 +16,7 @@
     </testsuite>
 
 {{- if .MisconfSummary }}
-    <testsuite tests="{{ add .MisconfSummary.Successes .MisconfSummary.Failures }}" failures="{{ .MisconfSummary.Failures }}" name="{{  .Target }}" errors="0" skipped="{{ .MisconfSummary.Exceptions }}" time="">
+    <testsuite tests="{{ add .MisconfSummary.Successes .MisconfSummary.Failures }}" failures="{{ .MisconfSummary.Failures }}" name="{{  .Target }}" errors="0" time="">
```
Comment (DmitriyLewen) [inline contrib/junit.tpl:19]: can we add `skipped="0"` instead of `.MisconfSummary.Exceptions`?

File: pkg/flag/misconf_flags.go (hunk 1 above)
Comment (simar7) [reply pkg/flag/misconf_flags.go]: fixed

File: contrib/junit.tpl (hunk 2 above)
Comment (nikpivkin) [reply contrib/junit.tpl:19]: Essentially we still have the option to ignore misconfigurations. Wouldn't `skipped=“0”` be confusing?

Comment (DmitriyLewen) [reply contrib/junit.tpl:19]: yes, you are right.
I was confused that we use `skipped="0"` for vulnerabilities.

Looks like we need to update our logic for `junit` template (in another PR of course)

File: pkg/rpc/convert.go
Comment (simar7) [reply pkg/rpc/convert.go:1]: thanks! I updated it.

**Review by nikpivkin — APPROVED** [review]

**Review by DmitriyLewen — APPROVED** [review]
lgtm

**Pull request merged**